### ISSUE TITLE
Build error when build python api from resource

### ISSUE THREAD

**Issue by fireFrappe** [openvinotoolkit/openvino#13036]
##### System information (version)

- OpenVINO=> 2022.1
- Operating System / Platform => windows11
- Compiler => vs2022
- Problem classification =>build error

##### Detailed description
Hi, when I want to build python api from resource follow https://github.com/openvinotoolkit/openvino/wiki/BuildingForWindows.
I used `` cmake -DENABLE_TESTS=ON -DENABLE_FUNCTIONAL_TESTS=ON -DENABLE_PYTHON=ON -DPYTHON_EXECUTABLE="C:\Users\Local_Admin\AppData\Local\Programs\Python\Python38\python.exe" -DPYTHON_LIBRARY="C:\Users\Local_Admin\AppData\Local\Programs\Python\Python38\libs\python38.lib" -DPYTHON_INCLUDE_DIR="C:\Users\Local_Admin\AppData\Local\Programs\Python\Python38\include" -DENABLE_OV_ONNX_FRONTEND=ON .. `` then `` cmake --build . --config Release -j 8 ``
but there were some wrong :
```
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(10,33): warning C4273: 'm_stat': inconsistent dll linkage [C:\openvino\build\src\bindings\python\tests\mock\mock_py_frontend\openvino_mock_py_frontend.
vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\include\mock_py_frontend/mock_py_frontend.hpp(613,19): message : see previous definition of 'private: static FeStat FrontEndMockPy::m_stat' [C:\openvino\build\src\bindings\pyth
on\tests\mock\mock_py_frontend\openvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(10,33): error C2491: 'FrontEndMockPy::m_stat': definition of dllimport static data member not allowed [C:\openvino\build\src\bindings\python\tests\mock
\mock_py_frontend\openvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(11,38): warning C4273: 'm_stat': inconsistent dll linkage [C:\openvino\build\src\bindings\python\tests\mock\mock_py_frontend\openvino_mock_py_frontend.
vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\include\mock_py_frontend/mock_py_frontend.hpp(429,22): message : see previous definition of 'private: static ModelStat InputModelMockPy::m_stat' [C:\openvino\build\src\bindings
\python\tests\mock\mock_py_frontend\openvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(11,38): error C2491: 'InputModelMockPy::m_stat': definition of dllimport static data member not allowed [C:\openvino\build\src\bindings\python\tests\mo
ck\mock_py_frontend\openvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(12,33): warning C4273: 'm_stat': inconsistent dll linkage [C:\openvino\build\src\bindings\python\tests\mock\mock_py_frontend\openvino_mock_py_frontend.
vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\include\mock_py_frontend/mock_py_frontend.hpp(91,22): message : see previous definition of 'private: static PlaceStat PlaceMockPy::m_stat' [C:\openvino\build\src\bindings\pytho
n\tests\mock\mock_py_frontend\openvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(12,33): error C2491: 'PlaceMockPy::m_stat': definition of dllimport static data member not allowed [C:\openvino\build\src\bindings\python\tests\mock\mo
ck_py_frontend\openvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(17,53): error C2491: 'GetAPIVersion': definition of dllimport function not allowed [C:\openvino\build\src\bindings\python\tests\mock\mock_py_frontend\o
penvino_mock_py_frontend.vcxproj]
C:\openvino\src\bindings\python\tests\mock\mock_py_frontend\src\mock_py_frontend.cpp(21,45): error C2491: 'GetFrontEndData': definition of dllimport function not allowed [C:\openvino\build\src\bindings\python\tests\mock\mock_py_frontend
\openvino_mock_py_frontend.vcxproj]
```

Any help? Thanks!

##### Issue submission checklist

 - [ ] I report the issue, it's not a question
   
 - [ ] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ ] There is reproducer code and related data files: images, videos, models, etc.

**Comment by ilya-lavrenov** [comment]
@praasz will it be fixed by https://github.com/openvinotoolkit/openvino/pull/12978 ?

**Comment by brmarkus** [comment]
The referenced Wiki page says:

> The software was validated on:
> Microsoft Windows 10 (64-bit) with Visual Studio 2019

Would you have a chance to do a quick test under a MS-Win10 and MS-VisualStudio 2019?

**Comment by ilya-lavrenov** [comment]
> The referenced Wiki page says:
> 
> > The software was validated on:
> > Microsoft Windows 10 (64-bit) with Visual Studio 2019
> 
> Would you have a chance to do a quick test under a MS-Win10 and MS-VisualStudio 2019?

I suppose this is compiler independent

**Comment by fireFrappe** [comment]
I think it is not the compiler problem, I disabled python before and can build successfully, this problem only occurred when I built python api.

**Comment by ilya-lavrenov** [comment]
@fireFrappe, I've attached the PR which potentially should fix the issue. Let's wait when it's merged to master or you can try it earlier from the PR directly.

**Comment by praasz** [comment]
> @praasz will it be fixed by #12978 ?

The linking of this `openvino_mock_py_frontend` library is fixed.
I've run build on MS VisualStudio 2019, Python3.9 and CMake options described in this bug with success.

**Comment by fireFrappe** [comment]
> > @praasz will it be fixed by #12978 ?
> 
> The linking of this `openvino_mock_py_frontend` library is fixed. I've run build on MS VisualStudio 2019, Python3.9 and CMake options described in this bug with success.

Which branch have you used? My branch is releases/2022/1

**Comment by praasz** [comment]
> > > @praasz will it be fixed by #12978 ?
> > 
> > 
> > The linking of this `openvino_mock_py_frontend` library is fixed. I've run build on MS VisualStudio 2019, Python3.9 and CMake options described in this bug with success.
> 
> Which branch have you used? My branch is releases/2022/1

It is on my fork in [this branch](https://github.com/praasz/openvino/tree/enable-mock-fe-tests-in-python-api-2.0-tests)
